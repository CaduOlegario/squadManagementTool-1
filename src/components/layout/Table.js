import React from 'react';
import MUIDataTable from "mui-datatables";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";
import Typography from "@material-ui/core/Typography";
import DeleteIcon from '@material-ui/icons/Delete';
import EditIcon from '@material-ui/icons/Edit';
import {
    createMuiTheme,
    MuiThemeProvider,
    withStyles
} from "@material-ui/core/styles";import Tooltip from "@material-ui/core/Tooltip";
import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import AddIcon from '@material-ui/icons/Add';
import useTheme from "@material-ui/core/styles/useTheme";

const getMuiTheme = (theme) =>
    createMuiTheme({
        ...theme,
        overrides: {
            MuiTable: {
                root: {
                    padding: "0 10px"
                }
            },
            MuiTableCell: {
                root: {
                    '&:hover': {
                        display: true,
                    }
                }
            },
            MuiTableRow: {
                root: {
                    '&:hover [class*="MuiTableCell-body"]': {
                        color: theme.palette.font.main,
                        backgroundColor: theme.palette.background.light
                    },
                },
            },
            MUIDataTable: {
                paper: {
                    padding: "0 20px",
                    height: "100%",
                    borderRadius: "14px"
                },
                responsiveScroll: {
                    maxHeight: 'none',
                }
            },
            MuiTableHead: {
                root: {
                    '&:hover .MuiTableCell-head': {
                        color: "black !important"
                    }
                }
            },
            MUIDataTableHeadCell: {
                data: {
                    fontWeight: 600
                }
            },
            MuiToolbar: {
                root: {
                    padding: "14px 16px 0 !important"
                }
            },
            MUIDataTableToolbar: {
              left: {
                  fontSize: "25px",
                  fontWeight: 600,
                  color: theme.palette.secondary.main
              }
            },
            MuiTypography: {
                subtitle2: {
                    fontWeight: 600
                }
            },
            MuiSvgIcon: {
                root: {
                    padding: "3px",
                    fontSize: "1.2rem"
                }
            },
            MuiButton: {
                root: {
                    padding: "5px 0",
                    minWidth: "40px",
                    borderRadius: "12px",
                    background: "linear-gradient(to bottom, " + theme.palette.primary.main + ", " + theme.palette.secondary.main + ")"
                },
                contained: {
                    color: "white",
                    fontSize: "1.7rem"
                }
            }
        }
    });

const styles = {
    tableRow: {
        '&:hover': {
            '& $actions': {
                opacity: 1
            }
        }
    },
    actions: {
        opacity: 0,
        cursor: "pointer"
    },
};

const customRowRenderLogic = (data, classes) => {
    return (
        <TableRow className={classes.tableRow}>
            <TableCell>
                <Typography variant="subtitle2">{data[0]}</Typography>
            </TableCell>
            <TableCell>
                <Box display="flex" flexDirection="row" alignItems="center">
                    <Box width="100%">
                        <Typography variant="subtitle2" component="span" >{data[1]}</Typography>
                    </Box>
                    <Box display="flex" className={classes.actions} alignItems="center">
                        <Tooltip title="Delete" placement="top" aria-label="delete">
                            <DeleteIcon className={classes.actionIcon}></DeleteIcon>
                        </Tooltip>
                        <Tooltip title="Edit" placement="top" aria-label="Edit">
                            <EditIcon className={classes.actionIcon}></EditIcon>
                        </Tooltip>
                    </Box>
                </Box>
            </TableCell>
        </TableRow>
    );
};

const Table = (props) => {
    const {classes} = props;
    const theme = useTheme();

    const dataTableProps = {
        ...props,
        title: <div>
            {props.title}
        </div>,
        options: {
            rowsPerPage: 7,
            rowsPerPageOptions: [7],
            print: false,
            filter: false,
            download: false,
            search: false,
            viewColumns: false,
            rowHover: false,
            selectableRows: "none",
            responsive: "vertical",
            tableBodyHeight: `calc(100% - 141px)`,
            tableBodyMaxHeight: '100%',
            elevation: 1,
            title: <div>
                <Typography variant="h1"> {props.title} </Typography>
            </div>,
            customRowRender: (data) => customRowRenderLogic(data, classes),
            customToolbar: ({displayData}) => {
              return <div>
                  <Button variant="contained">
                      <AddIcon style={{fontSize: "1.5rem"}}/>
                  </Button>
              </div>
            },
            ...props.options,
        }
    };

    return (
        <MuiThemeProvider theme={getMuiTheme(theme)}>
            <MUIDataTable {...dataTableProps} />
        </MuiThemeProvider>
    );
}

export default withStyles(styles)(Table);