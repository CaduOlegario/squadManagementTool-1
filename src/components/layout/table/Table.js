import React, {useState} from 'react';
import MUIDataTable from "mui-datatables";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";
import Typography from "@material-ui/core/Typography";
import DeleteIcon from '@material-ui/icons/Delete';
import EditIcon from '@material-ui/icons/Edit';
import {createMuiTheme, MuiThemeProvider, withStyles} from "@material-ui/core/styles";
import Tooltip from "@material-ui/core/Tooltip";
import Box from "@material-ui/core/Box";
import AddIcon from '@material-ui/icons/Add';
import useTheme from "@material-ui/core/styles/useTheme";
import {useHistory} from "react-router-dom";
import PropTypes from "prop-types";
import Button from "../button/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogTitle from "@material-ui/core/DialogTitle";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogActions from "@material-ui/core/DialogActions";
import {default as MaterialButton} from "@material-ui/core/Button";
import {useSnackbar} from "react-simple-snackbar";
import defaultToastConfig from "utils/ToastUtils";
import {HandleDeleteApi} from "utils/apiUtils";
import TeamApi from "api/TeamApi";


const getMuiTheme = (theme) =>
    createMuiTheme({
        ...theme,
        overrides: {
            MuiTable: {
                root: {
                    padding: "0 10px"
                }
            },
            MuiTableCell: {
                root: {
                    '&:hover': {
                        display: true,
                    }
                }
            },
            MuiTableRow: {
                root: {
                    '&:hover [class*="MuiTableCell-body"]': {
                        color: theme.palette.font.main,
                        backgroundColor: theme.palette.background.light
                    },
                },
            },
            MUIDataTable: {
                paper: {
                    height: "100%",
                    borderRadius: "14px"
                },
                responsiveBase: {
                    padding: "0 20px",
                    height: "82% !important",
                    overflowY: "auto",
                    '&::-webkit-scrollbar': {
                        width: '0'
                    },
                    '&::-webkit-scrollbar-track': {
                        boxShadow: 'inset 0 0 6px rgba(0,0,0,0.00)',
                        webkitBoxShadow: 'inset 0 0 6px rgba(0,0,0,0.00)'
                    },
                    '&::-webkit-scrollbar-thumb': {
                        backgroundColor: 'rgba(0,0,0,.1)',
                        outline: '1px solid slategrey'
                    }
                },
            },
            MuiTableHead: {
                root: {
                    '&:hover .MuiTableCell-head': {
                        color: "black !important"
                    }
                }
            },
            MUIDataTableHeadCell: {
                data: {
                    fontWeight: 600
                }
            },
            MuiToolbar: {
                root: {
                    padding: "9px 35px 13px !important",
                    borderBottom: "1px solid #ececec",
                }
            },

            MuiTypography: {
                subtitle2: {
                    fontWeight: 600
                }
            },
            MuiSvgIcon: {
                root: {
                    padding: "3px",
                    fontSize: "1.2rem"
                }
            },
        }
    });

const styles = {
    tableRow: {
        '&:hover': {
            '& $actions': {
                opacity: 1
            }
        }
    },
    actions: {
        opacity: 0,
        cursor: "pointer"
    },
};

/** Custom logic to render each row with actions */
const customRowRenderLogic = (data, classes, setDeleteDialogOpen, setSelectedTableData) => {
    return (
        <TableRow className={classes.tableRow}>
            <TableCell>
                <Typography variant="subtitle2">{data[1]}</Typography>
            </TableCell>
            <TableCell>
                <Box display="flex" flexDirection="row" alignItems="center">
                    <Box width="100%">
                        <Typography variant="subtitle2" component="span" >{data[2]}</Typography>
                    </Box>
                    <Box display="flex" className={classes.actions} alignItems="center">
                        <Tooltip title="Delete" placement="top" aria-label="delete">
                            <DeleteIcon
                                className={classes.actionIcon}
                                onClick={() => {
                                    setSelectedTableData(data);
                                    setDeleteDialogOpen(true);
                                }}
                            ></DeleteIcon>
                        </Tooltip>
                        <Tooltip title="Edit" placement="top" aria-label="Edit">
                            <EditIcon className={classes.actionIcon}></EditIcon>
                        </Tooltip>
                    </Box>
                </Box>
            </TableCell>
        </TableRow>
    );
};

/** Function to call delete api */
const deleteTeamApi = (id, openSnackbar, loader) => {
    const handleOpenSnackbar = (isApiSuccess) => {
        if(isApiSuccess) {
            openSnackbar("Team deleted successfully");
        } else {
            openSnackbar("Team couldn't be deleted");
        }
    };
    return HandleDeleteApi(TeamApi.deleteTeam, id, loader, handleOpenSnackbar);
};

/** Dialog for deleting a team */
const deleteDialog = (selectedTableData, isDeleteDialogOpen, setDeleteDialogOpen, openSnackbar, loader) => {

    if(!selectedTableData) return;

    const handleClose = () => setDeleteDialogOpen(false);

    return (
        <Dialog
            open={isDeleteDialogOpen}
            onClose={handleClose}
            aria-labelledby="alert-dialog-title"
            aria-describedby="alert-dialog-description"
        >
            <DialogTitle id="alert-dialog-title">Team delete</DialogTitle>
            <DialogContent>
                <DialogContentText id="alert-dialog-description">
                    Are you sure you wish to delete {selectedTableData[1]}'s team?
                </DialogContentText>
            </DialogContent>
            <DialogActions>
                <MaterialButton onClick={handleClose}>
                    Cancel
                </MaterialButton>
                <MaterialButton onClick={() => {
                    deleteTeamApi(selectedTableData[0], openSnackbar, loader);
                    handleClose();
                }} color="primary" autoFocus>
                    Delete
                </MaterialButton>
            </DialogActions>
        </Dialog>
    )
};

/** Table component aligned with prototype styling.  Accepts the same props from [MUI-Datatables](https://github.com/gregnb/mui-datatables) */
const Table = (props) => {
    const {classes, title, options, showButton, loader, ...dataTablePropsRest} = props;
    const [isDeleteDialogOpen, setDeleteDialogOpen] = useState(false);
    const [selectedTableData, setSelectedTableData] = useState(null);
    const [openSnackbar] = useSnackbar(defaultToastConfig);
    const theme = useTheme();
    const history = useHistory();

    const dataTableProps = {
        ...dataTablePropsRest,
        title: <div>
            <Typography variant="h1"> {title} </Typography>
        </div>,
        options: {
            print: false,
            filter: false,
            download: false,
            search: false,
            viewColumns: false,
            rowHover: false,
            pagination: false,
            selectableRows: "none",
            responsive: "standard",
            fixedHeader: true,
            elevation: 1,
            customRowRender: (data) => customRowRenderLogic(data, classes, setDeleteDialogOpen, setSelectedTableData),
            customToolbar: ({displayData}) => {
                return <div>
                    {showButton && <Button variant="contained" onClick={() => history.push("/form")}>
                        <AddIcon style={{fontSize: "1.5rem"}}/>
                    </Button>}
                </div>
            },
            ...options,
        }
    };

    return (
        <MuiThemeProvider theme={getMuiTheme(theme)}>
            <MUIDataTable {...dataTableProps} />
            {deleteDialog(selectedTableData, isDeleteDialogOpen, setDeleteDialogOpen, openSnackbar, loader)}
        </MuiThemeProvider>
    );
};

Table.propTypes = {
    /** Table's title */
    title: PropTypes.string.isRequired,
    /** Data to be displayed on the table */
    data: PropTypes.array,
    /** Columns of data to be displayer */
    columns: PropTypes.array,
    /** Define if table will contain the button to redirect to a register page */
    showButton: PropTypes.bool,
    /** @ignore */
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(Table);