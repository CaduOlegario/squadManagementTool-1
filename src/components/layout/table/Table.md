Table example, with ordering and actions

```js padded
import SnackbarProvider from 'react-simple-snackbar'

    const data = [
        {
            id: 1,
            name: "Jose",
            age: "25"
        },
        {
            id: 2,
            name: "Marcos",
            age: "14"
        },
        {
            id: 3,
            name: "Luis",
            age: "83"
        },
    ];

    const columns = [
        {
            name: "id",
            options: {
                display: 'excluded',
                viewColumns: "false"
            }
        },
        {
            name: "name",
            label: "Name",
        }, {
            name: "age",
            label: "Age"
        }
    ];

    <SnackbarProvider>
        <Table
            title="My Teams"
            data={data}
            columns={columns}
        />
    </SnackbarProvider>

```