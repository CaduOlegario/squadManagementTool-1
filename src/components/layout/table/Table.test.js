import React from 'react';
import renderer from 'react-test-renderer';
import Table from "./Table";
import SnackbarProvider from 'react-simple-snackbar'

const data = [
    {
        id: 1,
        name: "Jose",
        age: "25"
    },
    {
        id: 2,
        name: "Marcos",
        age: "14"
    },
    {
        id: 3,
        name: "Luis",
        age: "83"
    },
];

const columns = [
    {
        name: "id",
        options: {
            display: 'excluded',
            viewColumns: "false"
        }
    },
    {
        name: "name",
        label: "Name",
    }, {
        name: "age",
        label: "Age"
    }
];

it('renders correctly normal input', () => {
    const tree = renderer
        .create(
            <SnackbarProvider>
                <Table
                    title="My Teams"
                    data={data}
                    columns={columns}
                    loader={() => {
                    }}/>
            </SnackbarProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
