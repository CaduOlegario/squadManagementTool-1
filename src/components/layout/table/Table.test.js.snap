// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly normal input 1`] = `
<div
  className="MuiPaper-root MUIDataTable-paper-4 undefined MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    aria-label="Table Toolbar"
    className="MuiToolbar-root MuiToolbar-regular MUIDataTableToolbar-root-15 MuiToolbar-gutters"
    role="toolbar"
  >
    <div
      className="MUIDataTableToolbar-left-17"
    >
      <div>
        <h1
          className="MuiTypography-root MuiTypography-h1"
        >
           
          My Teams
           
        </h1>
      </div>
    </div>
    <div
      className="MUIDataTableToolbar-actions-19"
    >
      <div />
    </div>
  </div>
  <div
    className="MUIDataTableFilterList-root-30"
  />
  <div
    className="MUIDataTable-responsiveBase-7"
    style={
      Object {
        "height": "auto",
        "position": "relative",
      }
    }
  >
    <table
      className="MuiTable-root MUIDataTable-tableRoot-6"
      role="grid"
      tabIndex="0"
    >
      <caption
        className="MUIDataTable-caption-13"
      >
        <div>
          <h1
            className="MuiTypography-root MuiTypography-h1"
          >
             
            My Teams
             
          </h1>
        </div>
      </caption>
      <thead
        className="MuiTableHead-root MUIDataTableHead-main-32"
        role={null}
      >
        <tr
          className="MuiTableRow-root MUIDataTableHeadRow-root-36 MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-48 MUIDataTableHeadCell-fixedHeader-49"
            data-colindex={1}
            onMouseDown={[Function]}
            scope="col"
          >
            <span
              className="MUIDataTableHeadCell-toolButton-57"
              data-testid="headcol-1"
              onClick={[Function]}
              onKeyUp={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                aria-describedby={null}
                className="MUIDataTableHeadCell-sortAction-53"
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                title="Sort"
              >
                <div
                  className="MUIDataTableHeadCell-data-52"
                >
                  Name
                </div>
                <div
                  className="MUIDataTableHeadCell-sortAction-53"
                >
                  <span
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-55"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  />
                </div>
              </div>
            </span>
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-48 MUIDataTableHeadCell-fixedHeader-49"
            data-colindex={2}
            onMouseDown={[Function]}
            scope="col"
          >
            <span
              className="MUIDataTableHeadCell-toolButton-57"
              data-testid="headcol-2"
              onClick={[Function]}
              onKeyUp={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                aria-describedby={null}
                className="MUIDataTableHeadCell-sortAction-53"
                onBlur={[Function]}
                onFocus={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                onTouchEnd={[Function]}
                onTouchStart={[Function]}
                title="Sort"
              >
                <div
                  className="MUIDataTableHeadCell-data-52"
                >
                  Age
                </div>
                <div
                  className="MUIDataTableHeadCell-sortAction-53"
                >
                  <span
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-55"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  />
                </div>
              </div>
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root Table-tableRow-1"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle2"
            >
              Jose
            </h6>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <div
              className="MuiBox-root MuiBox-root-64"
            >
              <div
                className="MuiBox-root MuiBox-root-65"
              >
                <span
                  className="MuiTypography-root MuiTypography-subtitle2"
                >
                  25
                </span>
              </div>
              <div
                className="MuiBox-root MuiBox-root-66 Table-actions-2"
              >
                <svg
                  aria-describedby={null}
                  aria-hidden={true}
                  aria-label="delete"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="Delete"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
                <svg
                  aria-describedby={null}
                  aria-hidden={true}
                  aria-label="Edit"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="Edit"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </div>
            </div>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root Table-tableRow-1"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle2"
            >
              Marcos
            </h6>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <div
              className="MuiBox-root MuiBox-root-67"
            >
              <div
                className="MuiBox-root MuiBox-root-68"
              >
                <span
                  className="MuiTypography-root MuiTypography-subtitle2"
                >
                  14
                </span>
              </div>
              <div
                className="MuiBox-root MuiBox-root-69 Table-actions-2"
              >
                <svg
                  aria-describedby={null}
                  aria-hidden={true}
                  aria-label="delete"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="Delete"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
                <svg
                  aria-describedby={null}
                  aria-hidden={true}
                  aria-label="Edit"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="Edit"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </div>
            </div>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root Table-tableRow-1"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle2"
            >
              Luis
            </h6>
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
          >
            <div
              className="MuiBox-root MuiBox-root-70"
            >
              <div
                className="MuiBox-root MuiBox-root-71"
              >
                <span
                  className="MuiTypography-root MuiTypography-subtitle2"
                >
                  83
                </span>
              </div>
              <div
                className="MuiBox-root MuiBox-root-72 Table-actions-2"
              >
                <svg
                  aria-describedby={null}
                  aria-hidden={true}
                  aria-label="delete"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="Delete"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
                <svg
                  aria-describedby={null}
                  aria-hidden={true}
                  aria-label="Edit"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  title="Edit"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    aria-live="polite"
    className="MUIDataTable-liveAnnounce-14"
  />
</div>
`;
